app/controllers/compute_resources_controller.rb

```
@@ -49,6 +49,23 @@ def create
   def edit
   end
 
+  def associate
+    count = 0
+    @compute_resource.vms.each do |vm|
+      if Host.where(:uuid => vm.identity).empty?
+        host = @compute_resource.associated_host(vm.identity) if @compute_resource.respond_to?(:associated_host)
```

As this is called only for one CR, maybe move this check out of the loop and fail early if it doesn't respond to associated_host.